### PR TITLE
ft: ZENKO-740 add e2e tests crr metrics/healthcheck

### PR DESCRIPTION
**Pull request by philipyoo**
most additions are from package-lock file  x.x

### CODE REVIEW

File: tests/zenko_e2e/backbeat/tests/api/metrics.js [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+const { scalityS3Client, awsS3Client } = require('../../s3SDK');
+
+// Helpers
+const ReplicationUtility = require('../../ReplicationUtility');
+const { makeGETRequest, getResponseBody } = require('../../utils/request');
+
+const scalityUtils = new ReplicationUtility(scalityS3Client);
+const awsUtils = new ReplicationUtility(awsS3Client);
+
+// Vars
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/tests/api/metrics.js]: Minor: Are these comments needed?

File: tests/zenko_e2e/backbeat/tests/api/metrics.js [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+const srcBucket = `source-bucket-${Date.now()}`;
+const destBucket = process.env.AWS_S3_BACKBEAT_BUCKET_NAME;
+const hex = crypto.createHash('md5')
+    .update(Math.random().toString())
+    .digest('hex');
+const keyPrefix = `${srcBucket}/${hex}`;
+const key = `${keyPrefix}/object-to-replicate-${Date.now()}`;
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/tests/api/metrics.js]: Eventually I hope can move this key creation code into a helper util, but good for now! 😄

File: tests/zenko_e2e/backbeat/tests/api/metrics.js [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+            destBucket, roleArn, destLocation, next),
+    ], done));
+
+    afterEach(done => series([
+        next => scalityUtils.deleteVersionedBucket(srcBucket, next),
+        next => awsUtils.deleteAllVersions(destBucket,
+            `${srcBucket}/${keyPrefix}`, next),
+    ], done));
+
+    it('should return specific object properties', done => series([
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/tests/api/metrics.js]: No need for `series` here.

File: tests/zenko_e2e/backbeat/tests/api/metrics.js [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+    it('should get bucket-level throughput metrics', done => {
+        let versionId;
+        const destinationKey = `${srcBucket}/${key}`;
+
+        series([
+            next => scalityUtils.putObject(srcBucket, key, Buffer.alloc(10000),
+                next),
+            next => scalityUtils.compareObjectsAWS(srcBucket, destBucket, key,
+                undefined, next),
+            next => awsUtils.getObject(destBucket, `${destinationKey}`,
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/tests/api/metrics.js]: You can get the version ID from the callback of the initial `putObject` to save making a request to AWS here.

File: tests/zenko_e2e/backbeat/utils/request.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        path,
+    });
+    const req = http.request(options, res => cb(null, res));
+    req.on('error', err => cb(err));
+    req.end();
+}
+
+/**
+ * http request helper method
+ * @param {Object} path - url path
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/utils/request.js]: Should be of type String. Same for the above method `makeGETRequest`.

File: tests/zenko_e2e/backbeat/package.json [hunk 6]
```diff
@@ -16,6 +16,11 @@
     "mocha": "2.3.4",
     "request": "^2.87.0"
   },
+  "devDependencies": {
+    "eslint": "2.13.1",
```
Comment (bennettbuchanan) [inline tests/zenko_e2e/backbeat/package.json]: Maybe adding eslint should be another PR round, since it's not being used here.

Comment (philipyoo) [reply tests/zenko_e2e/backbeat/package.json]: sounds good, I added it because my atom editor was complaining and I couldn't figure out an alternative lol
